deploy/docs/Best_Practices.md

```diff
@@ -784,3 +784,63 @@ metadata:
         value: '12345'
 ```
+
+## OpenTelemetry queueing and batching
+
+OpenTelemetry goes with several parameters related to queue management.
+
+For [batch processor][batch_processor]:
+
+- `send_batch_size` defines the number of items (logs, metrics, traces) in one batch before it's sent further down the pipeline.
+- `timeout` defines time after which the batch is send regardless of the size (can be lower than `send_batch_size`).
+- `send_batch_max_size` is an upper limit of the batch size.
+
+_We could say that `send_batch_size` is soft limit and `send_batch_max_size` is hard limit of the batch size._
+
+For [sumologic exporter][sumologic_exporter]:
+
+- `max_request_body_size` defines maximum size of requests to sumologic before compression.
+- `timeout` defines connection timeout. It is recommended to adjust this value in relation to `max_request_body_size`.
+
+- `sending_queue.num_consumers` is the number of consumers that dequeue batches. It translates to maximum number of parallel connections to the sumologic backend.
+- `sending_queue.queue_size` is capacity of the queue in terms of batches (batches can vary between `1` and `send_batch_max_size`)
+
+**As effective value of `sending_queue.queue_size` depends on current traffic,**
+**there is no way to figure out optimal PVC size in relation to `sending_queue.queue_size`.**
+**Due to that, we recommend to set `sending_queue.queue_size` to high value in order to use maximum resources of PVC.**
+
+[batch_processor]: https://github.com/open-telemetry/opentelemetry-collector/tree/v0.44.0/processor/batchprocessor#batch-processor
+[sumologic_exporter]: https://github.com/SumoLogic/sumologic-otel-collector/tree/v0.0.50-beta.0/pkg/exporter/sumologicexporter#sumo-logic-exporter
+
+### Compaction
+
+The OpenTelemetry Collector doesn't have compaction mechanism.
+It reuses persistent storage space which but is unable to free it to filesystem.
+This leads to situation when database file can grow a lot (due to a spike in data traffic)
+but after some time only small piece of the file will be used for data storage (until next spike).
+
+### Examples
+
+Here is some useful examples and calculations for queue and batch parameters.
+
+For the calculations below we made an assumption that a single metric data point is around 1 kilobyte in size, including metadata.
+This assumption is based on the average data we ingest.
+Persistent storage doesn't compress data so we assume that single metric data point takes 1 kilobyte on disk as well.
+
+#### Outage with huge metrics spike
+
+Let's consider a huge metrics spike in your network while connection to the Sumologic is down.
+Huge load means that batch processor is going to push batches due to `send_batch_max_size` instead of `timeout`.
+The reliability of the system can be calculated using the following formulas:
+
+- If limited by queue_size: `number_of_instances*send_batch_max_size*sending_queue.queue_size/load_in_DPM` minutes.
```

Comment:
What is `number_of_instances`? Could you add explanation to this doc?

Reply:
It's number of `otelcol-metric` pods. I will add it to doc